packages/proxy/lib/http/util/cookies.ts

```diff
@@ -10,6 +10,47 @@ interface RequestDetails {
   needsCrossOriginHandling: boolean
 }
 
+/**
+ * Whether or not a url's scheme, domain, and top-level domain match to determine whether or not
+ * a cookie is considered first-party. See https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies#third-party_cookies
+ * for more details.
+ * @param {string} url1 - the first url
+ * @param {string} url2 - the second url
+ * @returns {boolean} whether or not the URL Scheme, Domain, and TLD match. This is called same-site and
+ * is allowed to have a different port or subdomain. @see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Sec-Fetch-Site#directives
+ * for more details.
+ */
+export const areUrlsSameSite = (url1: string, url2: string) => {
+  if (!url1 || !url2) return false
+
+  const { port: port1, ...parsedUrl1 } = cors.parseUrlIntoDomainTldPort(url1)
+  const { port: port2, ...parsedUrl2 } = cors.parseUrlIntoDomainTldPort(url2)
+
+  // If HTTPS, ports NEED to match. Otherwise, HTTP ports can be different and are same origin
+  const doPortsPassSameSchemeCheck = port1 !== port2 ? (port1 !== '443' && port2 !== '443') : true
+
+  return doPortsPassSameSchemeCheck && _.isEqual(parsedUrl1, parsedUrl2)
+}
+
+export const addCookieJarCookiesToRequest = (applicableCookieJarCookies: Cookie[] = [], requestCookieString: string[] = []): string => {
```

Comment:
```suggestion
export const addCookieJarCookiesToRequest = (applicableCookieJarCookies: Cookie[] = [], requestCookies: string[] = []): string => {
```

Reply:
addressed (somewhat) in [99d5ed2](https://github.com/cypress-io/cypress/pull/23438/commits/99d5ed23b824ec8206a6424edec54d010c18ec09). The name was used below so I went with `requestCookieStringArray`